pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -77,8 +80,82 @@ func (pl *CSILimits) Name() string {
 func (pl *CSILimits) EventsToRegister() []framework.ClusterEventWithHint {
 	return []framework.ClusterEventWithHint{
 		{Event: framework.ClusterEvent{Resource: framework.CSINode, ActionType: framework.Add}},
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}, QueueingHintFn: pl.isSchedulableAfterPodDeleted},
+	}
+}
+
+func (pl *CSILimits) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	deletedPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		logger.V(5).Info("Unexpected objects in isSchedulableAfterPodDeleted", "oldObj", oldObj, "err", err)
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPodDeleted: %w", err)
+	}
+
+	if deletedPod.Namespace != pod.Namespace {
+		return framework.Queue, nil
 	}
+
+	if len(deletedPod.Spec.Volumes) == 0 {
+		return framework.Queue, nil
+	}
+
+	csiNode, err := pl.csiNodeLister.Get(deletedPod.Spec.NodeName)
+	if err != nil {
+		logger.V(5).Info("Could not get a CSINode object", "nodeName", deletedPod.Spec.NodeName, "err", err)
+		return framework.Queue, fmt.Errorf("could not get a CSINode object: %w", err)
+	}
+
+	deletedPodDrivers := sets.Set[string]{}
+	for _, vol := range deletedPod.Spec.Volumes {
+		pvcName := ""
+		switch {
+		case vol.PersistentVolumeClaim != nil:
+			pvcName = vol.PersistentVolumeClaim.ClaimName
+		case vol.Ephemeral != nil:
+			pvcName = ephemeral.VolumeClaimName(pod, &vol)
+		default:
+			continue
+		}
+
+		pvc, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(pvcName)
+		if err != nil {
```

Comment:
What if not found? (I guess we should ignore not found error)

Reply:
49d395a7bc4c243655862865187436d9e4da886f